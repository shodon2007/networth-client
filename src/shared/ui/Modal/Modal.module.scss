.modal {
	left: 0;
	right: 0;
	top: 0;
	bottom: 0;
	position: fixed;
	z-index: -10;
	background: rgba(0, 0, 0, 0.5);
	opacity: 0;
	display: flex;
	align-items: center;
	justify-content: center;
	transition: opacity ease 0.3s;
	cursor: pointer;
}

.open {
	opacity: 1;
	z-index: 10;
}

.body {
	cursor: auto;
	width: 500px;
}
