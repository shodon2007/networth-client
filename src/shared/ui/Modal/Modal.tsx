import {FC, ReactNode} from "react";
import classNames from "src/shared/lib/classNames";

import cls from "./Modal.module.scss";

interface ModalProps {
	children: ReactNode;
	isOpen: boolean;
	close: () => void;
}

const Modal: FC<ModalProps> = ({children, isOpen, close}) => {
	return (
		<div
			onClick={close}
			className={classNames(cls.modal, {[cls.open]: isOpen})}
		>
			<div className={cls.body} onClick={(e) => e.stopPropagation()}>
				{children}
			</div>
		</div>
	);
};

export default Modal;
