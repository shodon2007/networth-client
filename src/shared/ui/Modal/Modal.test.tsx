import {fireEvent, render} from "@testing-library/react";
import Modal from "./Modal";

describe("testing Modal.tsx", () => {
	test("snapshoting modal", () => {
		const screen = render(
			<Modal isOpen={false} close={() => false}>
				hello worlds
			</Modal>,
		);
		const modal = screen.getByTestId("modal");
		expect(modal).toMatchSnapshot();
	});
	test("testing open, close modal", () => {
		let isOpen = true;
		const getModalComponent = () => (
			<Modal isOpen={isOpen} close={() => (isOpen = !isOpen)}>
				hello worlds
			</Modal>
		);
		const screen = render(getModalComponent());
		const modal = screen.getByTestId("modal");
		expect(modal).toHaveClass("open");
		fireEvent.click(modal);
		screen.rerender(getModalComponent());
		expect(modal).not.toHaveClass("open");
		fireEvent.click(modal);
		screen.rerender(getModalComponent());
		expect(modal).toHaveClass("open");
	});
});
