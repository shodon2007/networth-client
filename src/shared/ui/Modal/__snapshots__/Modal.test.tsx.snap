// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Modal.tsx snapshoting modal 1`] = `
<div
  class="modal"
  data-testid="modal"
>
  <div
    class="body"
  >
    hello worlds
  </div>
</div>
`;
